src/LanguageServer/Protocol/Workspaces/LspMiscellaneousFilesWorkspace.cs

```diff
@@ -63,6 +63,12 @@ internal sealed class LspMiscellaneousFilesWorkspace(ILspServices lspServices, I
             this, documentFilePath, sourceTextLoader, languageInformation, documentText.ChecksumAlgorithm, Services.SolutionServices, []);
         OnProjectAdded(projectInfo);
 
+        if (languageInformation.LanguageName == "Razor")
+        {
+            var docId = projectInfo.AdditionalDocuments.Single().Id;
+            return CurrentSolution.GetRequiredAdditionalDocument(docId);
+        }
+
         var id = projectInfo.Documents.Single().Id;
         return CurrentSolution.GetRequiredDocument(id);
```

Comment:
I kinda wish this was an else block so it was more visually clear it's the same logic, but don't touch it unless you have to touch the PR for some other reason.